core/src/main/java/com/datastrato/gravitino/storage/relational/service/RoleMetaService.java

```diff
@@ -145,8 +202,138 @@ public int deleteRoleMetasByLegacyTimeLine(long legacyTimeLine, int limit) {
                 SessionUtils.doWithoutCommitAndFetchResult(
                     GroupRoleRelMapper.class,
                     mapper ->
-                        mapper.deleteGroupRoleRelMetasByLegacyTimeLine(legacyTimeLine, limit)));
+                        mapper.deleteGroupRoleRelMetasByLegacyTimeLine(legacyTimeLine, limit)),
+        () ->
+            securableObjectsCount[0] =
+                SessionUtils.doWithoutCommitAndFetchResult(
+                    SecurableObjectMapper.class,
+                    mapper ->
+                        mapper.deleteSecurableObjectsByLegacyTimeLine(legacyTimeLine, limit)));
 
     return roleDeletedCount[0] + userRoleRelDeletedCount[0] + groupRoleRelDeletedCount[0];
   }
+
+  long getSecurableObjectEntityId(long metalakeId, String fullName, MetadataObject.Type type) {
+    if (fullName.equals(ALL_METALAKES) && type == MetadataObject.Type.METALAKE) {
+      return 0;
+    }
+
+    if (type == MetadataObject.Type.METALAKE) {
+      return MetalakeMetaService.getInstance().getMetalakeIdByName(fullName);
+    }
```

Comment:
I prefer to use `switch`, which is more readable

Reply:
If we use `switch`, we will have many duplicate code. 
Because we get a table, we need catalogId and schemaId. It's duplicate with the code to get a catalog and a schema.